plan/physical_plan_builder.go

```diff
@@ -48,13 +48,17 @@ var JoinConcurrency = 5
 func (p *DataSource) convert2TableScan(prop *requiredProperty) (*physicalPlanInfo, error) {
 	client := p.ctx.GetClient()
 	ts := PhysicalTableScan{
-		Table:               p.tableInfo,
-		Columns:             p.Columns,
-		TableAsName:         p.TableAsName,
-		DBName:              p.DBName,
-		physicalTableSource: physicalTableSource{client: client},
+		Table:       p.tableInfo,
+		Columns:     p.Columns,
+		TableAsName: p.TableAsName,
+		DBName:      p.DBName,
+		physicalTableSource: physicalTableSource{
+			client:          client,
+			NeedColHandle:   p.NeedColHandle,
+			unionScanSchema: p.unionScanSchema,
+		},
 	}.init(p.allocator, p.ctx)
-	ts.SetSchema(p.Schema())
+	ts.SetSchema(p.schema)
```

Comment:
Since no column is pushed into the schema, so we don't need to clone the schema.